src/Features/Core/Portable/GenerateType/AbstractGenerateTypeService.cs

```diff
@@ -121,7 +121,7 @@ protected AbstractGenerateTypeService()
                 }
             }
 
-            if (state.TypeToGenerateInOpt != null)
+            if (state.TypeToGenerateInOpt != null && !state.TypeToGenerateInOpt.MemberNames.Contains(state.Name))
```

Comment:
Thanks for the pr!

However, I would not do it this way.  With this, the feature will just silently not appear, which could be confusing.  Instead, a nicer way to do things is to offer the feature, but add the 'conflict' annotation on the appropriate node so that teh user knows there will be an issue.